nri_device_injector/nri_device_injector_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseDevices(t *testing.T) {
+	tests := map[string]struct {
+		container   string
+		annotations map[string]string
+		want        []device
+		wantErr     bool
+	}{
+		"No device annotations": {
+			container: "test",
+		},
+		"Unrelated device annotation": {
+			container:   "test",
+			annotations: map[string]string{"foo": "foo1"},
+		},
+		"One valid device annotation injecting to container": {
+			container: "test",
+			annotations: map[string]string{
+				"devices.gke.io/container.test": `
+- path: /dev/test
+`}, want: []device{{
+				Path: "/dev/test",
+			}},
+		},
+		"One valid device annotation injecting to pod": {
```

Comment:
@samuelkarp we had this conversation in the past, I think that for network devices this model will be more simple than existing one, IIUIC the only concern was that the OCI spec does not consider this devices? do you think that is a debate we can open?

Reply:
> IIUIC the only concern was that the OCI spec does not consider netdevices? do you think that is a debate we can open?

I think that'd be a reasonable discussion to have.  https://github.com/opencontainers/runtime-spec is the right place to open an issue.  https://github.com/opencontainers/runtime-spec/blob/main/config-linux.md describes what is supported both from a namespace (including network namespace) perspective and devices.

Reply:
opened https://github.com/opencontainers/runtime-spec/issues/1239